elastic/apm-server
# Add support for consuming OTLP/gRPC metrics

## Motivation/summary

Add support for consuming basic OTLP/gRPC metrics,
like we support in the OpenTelemetry Collector exporter.

We do not yet support distributions or summaries; these
will be added once the Elasticsearch enhancements we
are dependent on are available. If these are received, the
server will ignore them and increment a monitoring counter
to indicate that an unsupported metric was received and
dropped.

## Checklist

~- [ ] Documentation has been updated~ https://github.com/elastic/apm-server/issues/4678

## How to test these changes

1. Start apm-server with monitoring enabled
2. Run an application that is instrumented with OpenTelemetry to collect metrics (including some distributions and summaries), and configured to export OTLP to APM Server's standard host:port.
3. Repeat step 2, exporting to an OpenTelemetry Collector with the elastic exporter.
4. Compare the data received directly by APM Server to the data sent by the exporter.
5. Check the beats monitoring doc for apm-server, and check the value for `apm-server.otlp.grpc.metrics.consumer.unsupported_dropped` corresponds to the number of distribution and summary metrics sent

## Related issues

Closes #4503
